### DISCUSSION TITLE
[Question]: cobweb avoidance?

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#13316] [Ask A Question] [answered]
### What is your question?

Hello all

A general question: how might Frigate be able to help  minimise false alerts due to cobwebs?  

I note [this](https://github.com/blakeblackshear/frigate/discussions/11017) and [this](https://github.com/blakeblackshear/frigate/discussions/1128#discussioncomment-765747)  from three years ago but given the model developments observed since then I wondered if there are any refinements which have been implemented since 2021.

On my CCTV cameras,  night-time IR illumination causes a lot of retroreflection from cobwebs in the immediate vicinity of the lens, leading to lots of false alarms (these are untrained cameras i.e. no local "F+" model submitted _yet_): 

![image](https://github.com/user-attachments/assets/a4c1a39a-1d24-4191-9aa0-0e6a5ae273ec)

(1) Will a  Frigate+ model help with this problem?

(2) Other than increasing the "person" threshold to a value > 75% (according to the images above) what other methods, via the config, might reduce these false detections?

The latter option is not trivial because, for this camera, there are plenty of accurate daytime detections in the region of 71%/72% which of course is in the range which overlaps with the false night time detections. I have played a little bit with `improve_contrast` and perhaps it needs more work.

**Comment by hawkeye217** [comment, marked as answer]
As you noted, if there's accurate detections around the same confidence percentage, there isn't much that can be done (except perhaps clean off the cobwebs?). This is almost certainly a model related issue, so a properly trained Frigate+ model would probably help mitigate this problem.

**Reply by NickM-27** [reply]
Yeah, webs don't create false positives for me any more after the latest base model and some examples on my cameras

**Comment by ElectronicBattle** [comment]
Excellent - thanks for this information. It's interesting to note that the evidence developed over time is, and continues to be, that the trained model will very likely help (and, by the sounds of it, to quite a degree).  

How does it work in this instance - does the trained model establish what *is* a valid set of detections (car/person/licence plate) and therefore ignores what *isn't* (cobweb)? 

(My eight cameras are now all working and I am building up to the point where they all have enough images to work on, and then I can submit for paid F+.  Some of the eight only have a dozen images at the moment).

**Reply by NickM-27** [reply]
With the point where the base models are at, it sounds like you might have enough images for a first model. In my opinion it is pretty difficult to use 12 in a year, I have not had a single false positive in quite a long time

**Reply by ElectronicBattle** [reply]
I should get on with it, I think! Thanks - I'll apply in the next few days.
